### PR TITLE
feat: add function selector to pprof query

### PR DESCRIPTION
**Pull request by kolesnikovae**
The change extends pprof query interface with an option to scope the result to a specific function call sub-tree

### CODE REVIEW

File: api/types/v1/types.proto [hunk 1]
```diff
@@ -71,3 +71,11 @@ enum TimeSeriesAggregationType {
   TIME_SERIES_AGGREGATION_TYPE_SUM = 0;
   TIME_SERIES_AGGREGATION_TYPE_AVERAGE = 1;
 }
+
+message FunctionSelector {
+  repeated Function stack_trace = 1;
```
Comment (aleks-p) [inline api/types/v1/types.proto:76]: A question, would the user select a subtree of a stack trace or can they select functions that are not in the same call chain? Depending on the answer, `functions` could be a more flexible name for the field here.

File: pkg/phlaredb/symdb/resolver.go [hunk 2]
```diff
@@ -52,12 +67,15 @@ type lazyPartition struct {
 	done    chan struct{}
 }
 
-func NewResolver(ctx context.Context, s SymbolsReader) *Resolver {
+func NewResolver(ctx context.Context, s SymbolsReader, opts ...ResolverOption) *Resolver {
```
Comment (aleks-p) [inline pkg/phlaredb/symdb/resolver.go:70]: that is a nice way to make this more extensible, good stuff!

**Review by aleks-p — APPROVED** [review]
Looks good to me!

File: api/types/v1/types.proto (hunk 1 above)
Comment (kolesnikovae) [reply api/types/v1/types.proto:76]: The stack trace points to the subtree root. In the future we will likely extend the selector to cover the case when all the function instances (all the stack traces containing the function) are to be selected. This is the reason why `FunctionSelector` is extendable: we can always add a new field (e.g `function`) that would serve for that purpose.

However, I don't really like names `FunctionSelector` and `Function` here, although they are quite precise (as of now). I'm thinking about renaming it to `LocationSelector` and `Location` respectively

File: pkg/phlaredb/symdb/resolver.go (hunk 2 above)
Comment (kolesnikovae) [reply pkg/phlaredb/symdb/resolver.go:70]: Thanks to Rob Pike https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html :)

**Pull request merged**